cli/landscape.py

```diff
@@ -13,46 +13,126 @@
+        f"list of components to deploy. Possible Values: "
+        f"{', '.join([c.value for c in LandscapeComponent])}. By default, ALL components will "
+        f"be deployed.")
+    )=None,
+    webhook_dispatcher_chart_dir: CliHint(
+        typehint=str,
+        help="directory of webhook dispatcher chart",
+    )=None,
+    concourse_deployment_name: CliHint(
+        typehint=str, help="namespace and deployment name for Concourse"
+    )='concourse',
     timeout_seconds: CliHint(typehint=int, help="how long to wait for concourse startup")=180,
+    webhook_dispatcher_deployment_name: str='webhook-dispatcher',
     dry_run: bool=True,
 ):
-    '''Deploys a new concourse-instance using the given deployment name and config-directory.'''
-    which("helm")
+    '''Deploys the given components of the Concourse landscape.
+    '''
+    known_components = [component.value for component in LandscapeComponent]
+    # handle default (all known components)
+    if not components:
+        components = known_components
+    # Validate
+    unknown_components = [arg for arg in components if arg not in known_components]
+    if unknown_components:
+        raise ValueError(
+            f"Unknown components given: {', '.join(unknown_components)}. "
+            f"Known components: {', '.join(known_components)}"
+        )
+    if LandscapeComponent.WHD.value in components:
+        if not webhook_dispatcher_chart_dir:
+            raise ValueError(
+                f"--webhook-dispatcher-chart-dir must be given if component "
+                f"{LandscapeComponent.WHD.value} is to be deployed."
+            )
+        else:
+            webhook_dispatcher_chart_dir = existing_dir(webhook_dispatcher_chart_dir)
 
     _display_info(
         dry_run=dry_run,
         operation="DEPLOYED",
-        deployment_name=deployment_name,
+        deployment_name=concourse_deployment_name,
+        components=components,
     )
 
     if dry_run:
         return
 
+    if LandscapeComponent.SECRETS_SERVER.value in components:
+        info('Deploying Secrets Server')
+        deploy_secrets_server(
+            config_set_name=config_set_name,
+        )
+
+    if LandscapeComponent.CONCOURSE.value in components:
+        info('Deploying Concourse')
+        deploy_or_upgrade_concourse(
+            config_set_name=config_set_name,
+            deployment_name=concourse_deployment_name,
+            timeout_seconds=timeout_seconds,
+        )
+
+    if LandscapeComponent.WHD.value in components:
+        info('Deploying Webhook Dispatcher')
+        deploy_or_upgrade_webhook_dispatcher(
+            config_set_name=config_set_name,
+            chart_dir=webhook_dispatcher_chart_dir,
+            deployment_name=webhook_dispatcher_deployment_name,
+        )
+
+    if LandscapeComponent.MONITORING.value in components:
+        info('Deploying Monitoring stack')
+        deploy_or_upgrade_monitoring(
+            config_set_name=config_set_name,
+        )
+
```

Comment:
could be made more generic. as in: map enum values to deployment functions. it is sufficient to fallback to "deploying {enum_value}" - no need for friendiler (but custom) captions